core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -96,9 +90,11 @@ class CloudSqlInstance {
 
     if (authType == AuthType.IAM) {
       HttpRequestInitializer source = tokenSourceFactory.create();
-      this.iamAuthnCredentials = Optional.of(parseCredentials(source));
+      this.tokenSource = Optional.ofNullable(source);
+      this.accessTokenSupplier = new DefaultAccessTokenSupplier(tokenSource);
     } else {
-      this.iamAuthnCredentials = Optional.empty();
+      this.tokenSource = Optional.empty();
+      this.accessTokenSupplier = new DefaultAccessTokenSupplier(Optional.empty());
```

Comment:
Could we use a supplier that always returns an empty optional here?

Something like:

``` java
this.accessTokenSuppllier = () -> Optional.empty();
```